portfolio/src/main/java/com/google/sps/data/TimeRange.java

```diff
@@ -66,7 +66,67 @@ public static boolean equals(TimeRange a, TimeRange b) {
 
   /** Creates a {@code TimeRange} from {@code start} to {@code end}. */
   public static TimeRange fromStartEnd(Instant start, Instant end) {
-    return new TimeRange(
-        start, Duration.ofSeconds(end.getEpochSecond() - start.getEpochSecond() + 1));
+    return new TimeRange(start, Duration.ofSeconds(end.getEpochSecond() - start.getEpochSecond()));
+  }
+
+  /**
+   * Checks if this range completely contains another range. This means that {@code other} is a
+   * subset of this range. This is an inclusive bounds, meaning that if two ranges are the same,
+   * they contain each other.
+   */
+  public boolean contains(TimeRange otherRange) {
+    // If this range has no duration, it cannot contain anything.
+    if (duration.getSeconds() <= 0) {
+      return false;
+    }
+
+    // If the other range has no duration, then it is treated like a point that is
+    // anchored in its start time.
+    if (otherRange.duration.getSeconds() <= 0) {
+      return timeRangeContainsPoint(this, otherRange.start);
+    }
+
+    // Checks if the time range contains the other range's start and end points.
+    return timeRangeContainsPoint(this, otherRange.start)
+        && timeRangeContainsPoint(this, otherRange.end());
+  }
+
+  /**
+   * Checks if a time range contains a time point. Helper method for contains and overlaps. This
```

Comment:
If it is a helper method can we make it private, or package-private?

Reply:
Updated it to package-private.